src/core/symbolFactories.ts

```diff
@@ -9,36 +9,71 @@ import * as d3 from "d3";
  * A SymbolFactory is a function that takes in a symbolSize which is the edge length of the render area
  * and returns a string representing the 'd' attribute of the resultant 'path' element
  */
-export type SymbolFactory = (symbolSize: number) => string;
+export type SymbolFactory = (symbolSize: number, context?: CanvasRenderingContext2D) => string;
```

Comment:
This doesn't make sense from an API perspective - what are the semantics of the "context" argument? Who's supposed to use it? Also this won't always return a string anymore.

Actually this whole file never made sense - you invoked a function to return a new SymbolFactory, which you then invoked again with a symbolSize to return a path d string. What do you think?